common/v2/features/ReceiveAssets/ReceiveAssets.tsx

```diff
@@ -0,0 +1,255 @@
+  padding: 12px 0;
+  background: #f6f8fa;
+  text-align: center;
+`;
+
+const AssetFields = styled.div`
+  margin-bottom: 15px;
+  align-items: center;
+`;
+
+const Divider = styled.div`
+  height: 1px;
+  margin: 30px 0;
+  background: #e3edff;
+`;
+
+const FullWidthInput = styled(Input)`
+  width: 100%;
+`;
+
+const StyledSelect = styled(Select)`
+  width: 100%;
+  border-radius: 0.125em;
+  border: 0.125em solid rgba(63, 63, 68, 0.05);
+  outline: 0 0 0 0.25em rgba(0, 122, 153, 0.65);
+`;
+
+const Amount = styled.div`
+  width: 100%;
+`;
+
+const Asset = styled.div`
+  margin-top: 15px;
+  width: 100%;
+
+  .is-focused {
+    border-color: none;
+    outline: none;
+    box-shadow: 0 0 0 0.25em rgba(0, 122, 153, 0.65);
+  }
+`;
+
+const ErrorMessage = styled.span`
+  color: red;
+`;
+
+export function ReceiveAssets({ history }: RouteComponentProps<{}>) {
+  const { accounts } = useContext(AccountContext);
+  const { assets } = useContext(AssetContext);
+  const network = getNetworkByName(accounts[0].network);
+  const filteredAssets = network
+    ? assets
+        .filter(asset => asset.networkId === network.id)
+        .filter(asset => asset.type === 'base' || asset.type === 'erc20')
+    : [];
+  const assetOptions = filteredAssets.map(asset => ({ label: asset.name, id: asset.uuid }));
+  const [chosenAssetName, setAssetName] = useState(assetOptions[0].label);
+  const selectedAsset = filteredAssets.find(asset => asset.name === chosenAssetName);
+
+  const [requestAddress, setRequestAddress] = useState('');
+
+  const initialValues = {
+    amount: '0',
+    asset: { label: 'Ethereum', id: '7bbf42b1-9275-120b-0d0a-a788abd75ea0' },
+    chainId: network ? network.chainId : 1
+  };
+
+  const isValidAmount = (decimal: number) => (amount: string) =>
+    numberIsNotNegative(+amount) && validDecimal(amount, decimal);
+
+  const validateAmount = (amount: any) => {
+    let error;
+    if (selectedAsset) {
+      const { decimal } = selectedAsset;
+      if (decimal && !isValidAmount(decimal)(amount)) {
+        error = 'Please enter a valid amount';
+      }
+    }
+
+    return error;
+  };
+
+  return (
+    <ReceivePanel heading="Receive Assets" icon={receiveIcon} onBack={history.goBack}>
+      <Formik
+        initialValues={initialValues}
+        onSubmit={noop}
+        render={({ values: { amount, chainId }, errors }: FormikProps<typeof initialValues>) => (
+          <Form>
+            <Fieldset>
+              <SLabel htmlFor="recipientAddress">Recipient Address</SLabel>
+              <Field
+                name="recipientAddress"
+                component={({ field, form }: FieldProps) => (
+                  <AccountDropdown
+                    name={field.name}
+                    value={field.value}
+                    accounts={accounts}
+                    onSelect={(option: IExtendedAccount) => {
+                      form.setFieldValue(field.name, option);
+                      setRequestAddress(option.address);
```

Comment:
Formik already updates it's internal state with the new value on: `form.setFieldValue(field.name, option)`;
You can access it with `values` on line 141.
You don't need to keep an extra copy in the components state.